docs/html/user_guide.rst

```diff
@@ -772,6 +772,172 @@ archives are built with identical packages.
++------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| ``!=``     | Any version not equal to the specified version                                                                                                                        | **``!=3.1``:** any version other than ``3.1``                                                                                                                 |
++------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| ``~=``     | Any compatible release. Compatible releases are releases that are within the same major or minor version, assuming the package author is using semantic versioning.   | **``~=3.1``:** version ``3.1`` or later, but not version ``4.0`` or later. **``~=3.1.2``:** version ``3.1.2`` or later, but not version ``3.2.0`` or later.   |
++------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| ``*``      | Can be used at the end of a version number to represent "all"                                                                                                         | **``== 3.1.*``:** any version that starts with ``3.1``. Equivalent to ``~=3.1.0``.                                                                            |
++------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------+
+
+The detailed specification of supported comparison operators can be
+found in `PEP440 <https://www.python.org/dev/peps/pep-0440/>`__.
+
+Possible solutions
+------------------
+
+The solution to your error will depend on your individual use case. Here
+are some things to try:
+
+1. Audit your top level requirements
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+As a first step it is useful to audit your project and remove any
```

Comment:
This is a great point!

@pradyunsg @uranusjr @pfmoore Should we point out `pip freeze` here?

Reply:
I don't think it would help much, because:

1. It reports on what's installed, not what you're requesting.
2. It tightly version pins everything, which is the opposite of what we want here, I think.

We're really directing the user to look at requirements defined in their `setup.py` (or equivalent in whatever build tool they are using) so maybe we should explicitly say that?

Reply:
Thanks for your feedback.  Adding "e.g. from your `setup.py` or `requirements.txt` files" to this section.